pkg/apis/core/v1alpha1/disable_types.go

```diff
@@ -18,10 +18,27 @@ type ConfigMapDisableSource struct {
 }
 
 type DisableStatus struct {
-	// Whether this is currently disabled.
+	// Whether this is currently disabled. Deprecated in favor of `State`.
 	Disabled bool `json:"disabled" protobuf:"varint,1,opt,name=disabled"`
 	// The last time this status was updated.
 	LastUpdateTime metav1.Time `json:"lastUpdateTime" protobuf:"bytes,2,opt,name=lastUpdateTime"`
 	// The reason this status was updated.
 	Reason string `json:"reason" protobuf:"bytes,3,opt,name=reason"`
+	// Whether this is currently disabled (if known)
+	State DisableState `json:"state" protobuf:"bytes,4,opt,name=state,casttype=DisableState"`
```

Comment:
`DisableState` vs `DisableStatus` are not very discriminating names. I haven't come up with a better option here.